docs/src/tutorial/cython_tutorial.rst

```diff
@@ -255,49 +326,57 @@ sense for those lines to be yellow. Same for the list comprehension because
 it involves the creation of a Python object. But the line ``if n % i == 0:``, why?
 We can examine the generated C code to understand:
 
-.. figure:: python_division.png
+.. tabs::
+    .. group-tab:: Cython
+
+        .. figure:: python_division.png
+
+    .. group-tab:: Pure Python
+
+        .. figure:: python_division_pure.png
 
 We can see that some checks happen. Because Cython defaults to the
 Python behavior, the language will perform division checks at runtime,
 just like Python does. You can deactivate those checks by using the
 :ref:`compiler directives<compiler-directives>`.
 
-Now let's see if, even if we have division checks, we obtained a boost in speed.
-Let's write the same program, but Python-style:
+Now let's see if we get a speed increase even if there is a division check.
+Let's write the same program, but in Python:
 
 .. literalinclude:: ../../examples/tutorial/cython_tutorial/primes_python.py
+    :caption: primes_python.py / primes_python_compiled.py
 
-It is also possible to take a plain ``.py`` file and to compile it with Cython.
-Let's take ``primes_python``, change the function name to ``primes_python_compiled`` and
-compile it with Cython (without changing the code). We will also change the name of the
-file to ``example_py_cy.py`` to differentiate it from the others.
-Now the ``setup.py`` looks like this::
+It is possible to take a plain (unannotated) ``.py`` file and to compile it with Cython.
+Let's create a copy of ``primes_python`` and name it ``primes_python_compiled`` and
+compile it with Cython (without changing the code).
+Now the ``setup.py`` looks like this:
 
-    from setuptools import setup
-    from Cython.Build import cythonize
+.. tabs::
+    .. group-tab:: Cython
 
-    setup(
-        ext_modules=cythonize(['example.pyx',        # Cython code file with primes() function
-                               'example_py_cy.py'],  # Python code file with primes_python_compiled() function
-                              annotate=True),        # enables generation of the html annotation file
-    )
+        .. literalinclude:: ../../examples/tutorial/cython_tutorial/setup_primes.py
+
+    .. group-tab:: Pure Python
+
+        .. literalinclude:: ../../examples/tutorial/cython_tutorial/setup_primes_pure.py
```

Comment:
I'm not sure if it's worth putting these in external files. The main point here are the file names that get passed into `cythonize()`, and compiling the file won't validate them. It's easier to see that they are correct when inlined here.

Reply:
@scoder it's just not very convenient to change and maintain them if they are in a block of tabs. It is possible to remove .py from the name so that the tests do not occur.

Reply:
And look at [this file](https://github.com/cython/cython/blob/master/docs/examples/tutorial/cython_tutorial/setup.py) then, it was there before and everything was fine

Reply:
It just seems wrong to me to have these under a different name than `setup.py` and to have them in files at all, where they expose a higher risk of not matching the text. It's too easy to rename one of the other files and forget to adapt this one.

The fact that this has been done before does not mean that it was previously a good decision.

Reply:
> It's too easy to rename one of the other files and forget to adapt this one.

I cannot disagree. Putting this in separate files allows you to test them, but it probably makes very little sense, even more so compared to the possibility of forgetting to update them.
In general, I think I will agree, and in the near future I will add them as text.

Reply:
Yes, I think it would be better to have the setup.py scripts included as text, not files.

Reply:
resolved [here](https://github.com/cython/cython/pull/4226/commits/e1cb02468f09921fb8f019a1d0efc3558b26c0e6)